app/models/manageiq/providers/ibm_power_hmc/infra_manager/vm.rb

```diff
@@ -43,7 +47,15 @@ def raw_destroy
   def raw_suspend
   end
 
-  def raw_rename
+  def raw_rename(new_name)
+    ext_management_system.with_provider_connection do |connection|
+      connection.modify_object do
+        provider_object(connection).tap { |lpar| lpar.name = new_name }
+      end
+    rescue IbmPowerHmc::Connection::HttpError => e
+      $ibm_power_hmc_log.error("error renaming LPAR #{ems_ref} to #{new_name}: #{e}")
```

Comment:
Minor observation: This error could come from both Lpar and Vios, so this error message might be a little confusing. Same for the `lpar` variable on line 53 above.

Reply:
Yes, we are sometimes using the term LPAR as a generic term that also includes VIOSes (VIOSes being considered a special case of logical partition in this case, if that makes sense). I am not sure what generic term would fit best. Maybe just "partition"

Reply:
The HMC XML schema has a `BasePartition` type that is inherited by `VirtualIOServer` and `LogicalPartition`:
https://github.com/IBM/ibm_power_hmc_sdk_ruby/blob/e826b32fd5693169478f9a28218c3e2098f24df8/lib/ibm_power_hmc/parser.rb#L405

Reply:
Makes sense - thanks for the clarification!